### PR TITLE
DEVPROD-9629: quarantine unresponsive static hosts

### PR DESCRIPTION
**Pull request by Kimchelly**
DEVPROD-9629

### Description
I looked into the problematic release task, and it looks like task group would have been unstuck if the host was quarantined rather than manually modifying the last task group info on the host. Since quarantining is the standard solution to taking a broken static host out of the host pool, I decided to just make that more robust and add a mechanism to automatically quarantine long-unresponsive static hosts.

* When quarantining a host, always clear its running task as well.
* Add logic to the job that decommissions/terminates unresponsive EC2 hosts to also handle static hosts. If a static host hasn't communicated with Evergreen for over 2 hours, automatically quarantine it.
* Rename some (unrelated) host timeout constants for clarity.

### Testing
* Added unit tests for quarantine behavior.
* Manually tested in staging that quarantining a host running a task group allowed the task group to run on a different host on restart.

### Documentation
The ops guide already has instructions on quarantining static hosts. I added some instructions on how/when to unquarantine hosts in bulk.

### CODE REVIEW

File: units/host_monitoring_check.go [hunk 1]
```diff
@@ -92,10 +89,48 @@ func (j *hostMonitorExternalStateCheckJob) Run(ctx context.Context) {
...
+// because static hosts do not have a reliable way to verify if they're actually
+// healthy and able to run tasks. That means that in some rare cases, this could
+// produce false positives (e.g. if Evergreen is down for hours, the static
+// hosts can't reach it) and false negatives (because it waits a very long time
+// before declaring a host unhealthy).
+func (j *hostMonitorExternalStateCheckJob) handleUnresponsiveStaticHost(ctx context.Context) error {
+	if j.host.Provider != evergreen.ProviderNameStatic {
+		return nil
+	}
+	if j.host.Status == evergreen.HostQuarantined {
```
Comment (hadjri) [inline units/host_monitoring_check.go:112]: can we instead check if status is not in `evergreen.DownHostStatus`?

File: units/util.go [hunk 2]
```diff
@@ -30,32 +30,42 @@ func HandlePoisonedHost(ctx context.Context, env evergreen.Environment, h *host.
...
 	return catcher.Resolve()
 }
 
-func DisableAndNotifyPoisonedHost(ctx context.Context, env evergreen.Environment, h *host.Host, reason string) error {
+// DisableAndNotifyPoisonedHost disables an unhealthy host so that it cannot run
+// any more tasks, clears any tasks that have been straned on it, and enqueues a
+// job to notify that a host was disabled. If canDecommission is true and the
+// host is an ephemeral host, it will decommission the host instead of
+// quarantine it.
+func DisableAndNotifyPoisonedHost(ctx context.Context, env evergreen.Environment, h *host.Host, canDecommission bool, reason string) error {
```
Comment (hadjri) [inline units/util.go:49]: Can we make this function no-op if the task is already quarantined or decommissioned depending on the case?

**Review by hadjri — CHANGES_REQUESTED** [review]

File: units/util.go [hunk 3]
```diff
@@ -30,32 +30,42 @@ func HandlePoisonedHost(ctx context.Context, env evergreen.Environment, h *host.
...
-		catcher.Wrapf(DisableAndNotifyPoisonedHost(ctx, env, h, reason), "disabling poisoned host '%s'", h.Id)
+		catcher.Wrapf(DisableAndNotifyPoisonedHost(ctx, env, h, true, reason), "disabling poisoned host '%s'", h.Id)
 	}
 
 	return catcher.Resolve()
 }
 
-func DisableAndNotifyPoisonedHost(ctx context.Context, env evergreen.Environment, h *host.Host, reason string) error {
+// DisableAndNotifyPoisonedHost disables an unhealthy host so that it cannot run
+// any more tasks, clears any tasks that have been straned on it, and enqueues a
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply units/util.go]: Fixed.

File: units/util.go (hunk 2 above)
Comment (Kimchelly) [reply units/util.go:49]: L50-52 should already no-op when the host is unhealthy. Or did you mean something about the task?

File: units/util.go [hunk 4]
```diff
@@ -30,32 +30,42 @@ func HandlePoisonedHost(ctx context.Context, env evergreen.Environment, h *host.
...
 	if utility.StringSliceContains(evergreen.DownHostStatus, h.Status) {
 		return nil
 	}
 
-	err := h.DisablePoisonedHost(ctx, reason)
-	if err != nil {
-		return errors.Wrap(err, "disabling poisoned host")
+	if canDecommission && h.Provider != evergreen.ProviderNameStatic {
+		if err := h.SetDecommissioned(ctx, evergreen.User, true, reason); err != nil {
+			return errors.Wrap(err, "decommissioning host")
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply units/util.go]: Done.

File: units/host_monitoring_check.go (hunk 1 above)
Comment (Kimchelly) [reply units/host_monitoring_check.go]: Hmm yeah that works as well. Switched it to that.

**Review by Kimchelly — COMMENTED** [review]
I know it's approved, but re-requesting review on [this comment](https://github.com/evergreen-ci/evergreen/pull/8286/files#r1757170272).

File: units/util.go (hunk 2 above)
Comment (hadjri) [reply units/util.go:49]: Yeah disregard

**Review by hadjri — APPROVED** [review]

**Pull request merged**